tests/conftest.py

```diff
@@ -168,18 +171,44 @@ def hang_url(local_server: LocalHttpServer):
         local_server.hang_forever_stop()
 
 
-@pytest.fixture
-def bad_ssl_url():
-    """
-    Return a URL with an expired SSL certificate.
+@pytest.fixture(scope="session")
+def ssl_context_err_cert_authority_invalid():
+    """Return a SSL context with an invalid certificate authority."""
+    context = ssl.SSLContext(ssl.PROTOCOL_TLS_SERVER)
+
+    cert_file_name = Path(__file__).parent / "tools" / "cert.pem"
+    key_file_name = Path(__file__).parent / "tools" / "key.pem"
+
+    context.load_cert_chain(cert_file_name, key_file_name)
+
+    return context
+
 
+@pytest.fixture(scope="session")
+def ssl_httpserver_err_cert_authority_invalid(
```

Comment:
How about we call it `https_server_invalid_cert` ?

Reply:
I used a slightly different name taking your suggestion into account